### PR TITLE
refactor(): #CRRE-579 refactor using universal model to prepare the multi library

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Refactoring to use the universal order model for export flow

## Checklist tests
Check the CSS exports and the CSS of the requests sent to the bookseller.
The calculation of the VAT price was not correct.

## Issue ticket number and link
[CRRE-579](https://jira.support-ent.fr/browse/CRRE-579)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/crre/controllers/OrderController.java [hunk 1]
```diff
@@ -312,96 +281,6 @@ public void export(final HttpServerRequest request) {
...
-                newOffer.put(Field.ID, offer.getString(Field.EAN));
-                newOffer.put(Field.NAME, offer.getString(Field.TITRE));
-                if (gratuite > 0) {
-                    offers.add(newOffer);
-                }
-            }
-        }
-        return offers;
-    }
-
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/controllers/OrderController.java:404]: Ces fonctions ne sont plus utilisées.

File: src/main/java/fr/openent/crre/controllers/OrderController.java [hunk 2]
```diff
@@ -437,17 +316,6 @@ private static String generateExportLine(HttpServerRequest request, OrderUnivers
...
-        return (orderRegionBeautifyModel.getOrderRegion().getAmount() != null ? orderRegionBeautifyModel.getOrderRegion().getAmount() : "") + ";" +
-                (orderRegionBeautifyModel.getPriceht() != null ? orderRegionBeautifyModel.getPriceht() : "") + ";" +
-                (orderRegionBeautifyModel.getTva5() != null ? orderRegionBeautifyModel.getTva5() : "") + ";" +
-                (orderRegionBeautifyModel.getTva20() != null ? orderRegionBeautifyModel.getTva20() : "") + ";" +
-                (orderRegionBeautifyModel.getUnitedPriceTTC() != null ? convertPriceString(orderRegionBeautifyModel.getUnitedPriceTTC()) : "") + ";" +
-                (orderRegionBeautifyModel.getTotalPriceHT() != null ? convertPriceString(orderRegionBeautifyModel.getTotalPriceHT()) : "") + ";" +
-                (orderRegionBeautifyModel.getTotalPriceTTC() != null ? convertPriceString(orderRegionBeautifyModel.getTotalPriceTTC()) : "") + ";" +
-                (orderRegionBeautifyModel.getOrderRegion().getComment() != null ? orderRegionBeautifyModel.getOrderRegion().getComment().replaceAll("\n", "").replaceAll("\r", "") : "") + ";";
-    }
-
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/controllers/OrderController.java:450]: Ces fonctions ne sont plus utilisées.

File: src/main/java/fr/openent/crre/helpers/OrderHelper.java [hunk 3]
```diff
@@ -75,4 +112,80 @@ public static Future<List<OrderUniversalModel>> listOrderAndCalculatePrice(Filte
 
         return promise.future();
     }
+
+    //todo #Territoire Rajouter territoire
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/helpers/OrderHelper.java:116]: Todo pour les prochains tickets

File: src/main/java/fr/openent/crre/controllers/OrderRegionController.java [hunk 4]
```diff
@@ -300,6 +300,7 @@ public void getAllProjects(HttpServerRequest request) {
     }
 
 
+    // TODO: verif si a delete
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/controllers/OrderRegionController.java:303]: Todo pour les prochains tickets

File: src/main/java/fr/openent/crre/controllers/OrderRegionController.java [hunk 5]
```diff
@@ -450,6 +451,7 @@ public Future<Void> getOrderLDE(Handler<OrderLDEModel> orderLDEModelHandler) {
         return promise.future();
     }
 
+    // TODO: verif si a delete
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/controllers/OrderRegionController.java:454]: Todo pour les prochains tickets

File: src/main/java/fr/openent/crre/service/impl/DefaultOrderRegionService.java [hunk 6]
```diff
@@ -188,34 +188,6 @@ public Future<List<OrderRegionEquipmentModel>> getOrdersRegionByStatus(OrderStat
...
-                        .filter(JsonObject.class::isInstance)
-                        .map(JsonObject.class::cast)
-                        .map(jsonObject -> new OrderRegionComplex(new JsonObject(jsonObject.getString(Field.RESULT))))
-                        .collect(Collectors.toList());
-                promise.complete(orderRegionComplexList);
-            }
-        }));
-        return promise.future();
-    }
-
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/service/impl/DefaultOrderRegionService.java:218]: Ces fonctions ne sont plus utilisées.

File: src/main/java/fr/openent/crre/service/impl/DefaultOrderRegionService.java [hunk 7]
```diff
@@ -250,22 +222,6 @@ private String getQueryOrdersRegionById(boolean oldTable, List<Integer> idsOrder
...
-        String query = selectOrderRegion(oldTable);
-        query += "WHERE ore.id in " + Sql.listPrepared(idsOrder.toArray());
-        JsonArray params = new fr.wseduc.webutils.collections.JsonArray();
-        for (Integer id : idsOrder) {
-            params.add(id);
-        }
-        query = groupOrderRegion(oldTable, query);
-        Sql.getInstance().prepared(query, params, SqlResult.validResultHandler(arrayResponseHandler));
-    }
-
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/service/impl/DefaultOrderRegionService.java:268]: Ces fonctions ne sont plus utilisées.

File: src/main/java/fr/openent/crre/service/impl/DefaultOrderRegionService.java [hunk 8]
```diff
@@ -732,36 +676,6 @@ private TransactionElement deletedOrderList(List<Long> ordersClientIdList, Strin
...
-        });
-    }
-
-    @Override
-    public void getStatusByOrderId(Handler<Either<String, JsonArray>> arrayResponseHandler) {
-        String query = "SELECT id FROM " + Crre.crreSchema + ".\"order-region-equipment-old\" WHERE owner_id != 'renew2021-2022'";
-        Sql.getInstance().prepared(query, new JsonArray(), SqlResult.validResultHandler(arrayResponseHandler));
-
-    }
-
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/service/impl/DefaultOrderRegionService.java:764]: Ces fonctions ne sont plus utilisées.

File: src/main/java/fr/openent/crre/service/impl/DefaultOrderRegionService.java [hunk 9]
```diff
@@ -812,182 +726,25 @@ public Future<Map<ProjectModel, List<OrderRegionEquipmentModel>>> getOrderRegion
...
-    @Override
-    public JsonObject generateExport(List<OrderRegionBeautifyModel> logs) {
-        StringBuilder report = new StringBuilder(UTF8_BOM).append(UTF8_BOM).append(getExportHeader());
-        HashSet<String> structures = new HashSet<>();
-        for (OrderRegionBeautifyModel orderRegionBeautifyModel: logs) {
-            report.append(generateExportLine(orderRegionBeautifyModel));
-            if (!StringUtils.isEmpty(orderRegionBeautifyModel.getUaiStructure())) {
-                structures.add(orderRegionBeautifyModel.getUaiStructure());
-            }
-        }
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/service/impl/DefaultOrderRegionService.java:987]: Ces fonctions ne sont plus utilisées.

**Review by flmartineau — APPROVED** [review]

File: src/main/java/fr/openent/crre/helpers/OrderHelper.java [hunk 10]
```diff
@@ -75,4 +112,80 @@ public static Future<List<OrderUniversalModel>> listOrderAndCalculatePrice(Filte
...
+                    }
+                } else if (lep.getJsonArray(Field.CONDITIONS).size() == 1) {
+                    gratuit = lep.getJsonArray(Field.CONDITIONS).getJsonObject(0).getInteger(Field.CONDITIONS_FREE);
+                    gratuite = (int) (lep.getJsonArray(Field.CONDITIONS).getJsonObject(0).getInteger(Field.FREE) * Math.floor(amount * 1.0 / gratuit));
+                }
+
+                if (gratuite > 0) {
+                    orderUniversalOffer.setAmount(gratuite);
+                    orderUniversalOffer.setName(lep.getString(Field.TITRE));
+                    orderUniversalOffer.setTitre("Manuel " + offersJsonArray.getJsonObject(0).getJsonArray(Field.LICENCE).getJsonObject(0).getString(Field.VALEUR));
```
[the remark this answers is not in the document]
Comment (Sofianebernoussi) [reply src/main/java/fr/openent/crre/helpers/OrderHelper.java:175]: il va être supprimer ce "Manuel"

**Review by Sofianebernoussi — APPROVED** [review]

**Pull request merged**